src/systemd-sonic-generator/systemd-sonic-generator.c

```diff
@@ -501,22 +655,31 @@ static int install_unit_file(char* unit_file, char* target, char* install_dir) {
         for (int i = 0; i < num_asics; i++) {
 
             if (strstr(target, "@") != NULL) {
-                target_instance = insert_instance_number(target, i);
+                target_instance = insert_instance_number(target, i, "");
             }
             else {
                 target_instance = strdup(target);
             }
 
-            r = create_symlink(unit_file, target_instance, install_dir, i);
+            r = create_symlink(unit_file, target_instance, install_dir, i, "");
             if (r < 0)
                 fprintf(stderr, "Error installing %s for target %s\n", unit_file, target_instance);
 
             free(target_instance);
 
         }
-    }
-    else {
-        r = create_symlink(unit_file, target, install_dir, -1);
+    } else if (num_dpus > 0 && strstr(unit_file, "@") != NULL) {
+        // If multi-instance service for DPU
+        // Install each DPU units to the host main instance only,
+        // E.g. install database@dpu0.service, database@dpu1.service to multi-user.target.wants
+        // We don't have case like to install xxx@dpu0.service to swss@dpu0.service.wants
+        for (int i = 0; i < num_dpus; i++) {
+            r = create_symlink(unit_file, target, install_dir, i, DPU_PREFIX);
+            if (r < 0)
+                fprintf(stderr, "Error installing %s for target %s\n", unit_file, target);
```

Comment:
No error handling is needed here?

Reply:
Here just follows the original style that no any error handling after create_symlink error. And I don't feel we have an opportunity to handle this error. So, just to expose all create_symlink isn't a bad idea.